opennms-services/src/main/java/org/opennms/netmgt/collectd/PointerLikeIndexPropertyExtender.java

```diff
@@ -0,0 +1,155 @@
+            return null;
+        }
+        
+        final String indexInSourceTable = getIndirectIndex(sourceAttributes,  targetResource , pointerIndex);
+        if (indexInSourceTable != null) {
+            final SnmpValue value = getValueInSourceTable(sourceAttributes, indexInSourceTable, sourceType, sourceAlias);           
+            if (value != null) {
+    			final MibPropertyAttributeType type = getAttributeType(targetResource,property);
+            	return new SnmpAttribute(targetResource, type, value); 
+            }
+        }
+        return null;
+    }
+    
+	final private MibPropertyAttributeType getAttributeType(SnmpCollectionResource targetResource, MibObjProperty property) {
+		final AttributeGroupType	groupType 		= targetResource.getGroupType(property.getGroupName());
+		final ResourceType 			resourceType 	= targetResource.getResourceType();
+		
+		return new MibPropertyAttributeType(resourceType, property, groupType);
+	}
+
+	final private String getIndirectIndex(List<CollectionAttribute> sourceAttributes, SnmpCollectionResource targetResource,
+			String pointerIndex) {
+    	final String currentSourceTypeName	= targetResource.getResourceTypeName();
+		final String currentInstance		= targetResource.getInstance();
+		
+    	final Optional<CollectionAttribute> target 	=  
+    			sourceAttributes.stream().filter(a -> matches(currentSourceTypeName,pointerIndex , currentInstance, a)).findFirst();
+    	if ((target == null) ||  (! target.isPresent())){
+    		LOG.warn("Unable to find an indirect value in resource {} for index {} and instance {}", currentSourceTypeName, pointerIndex, currentInstance);
+    		return null;
+    	}
+    	String result = target.get().getStringValue();
+        LOG.debug("{} value {} points to {}  ",pointerIndex, currentInstance, result);
+
+    	return result;
+	}
+	private SnmpValue getValueInSourceTable(List<CollectionAttribute> sourceAttributes, String indexInSourceTable,
+			String sourceType, String sourceAlias) {
+		final Optional<CollectionAttribute> target = 
+				sourceAttributes.stream().filter(a -> matchesValue(sourceType, sourceAlias, indexInSourceTable, a)).findFirst();
+		
+		if ((target == null) ||  (! target.isPresent())){
+        	LOG.debug("Unable to find a value in resource  {} for index {} and instance {}", sourceType, sourceAlias, indexInSourceTable);
+        	return null;
+		}
+		final  SnmpValue result = 	getSnmpValue(target);
+		LOG.debug("Value in source table is" + result);
+		return result;
+	}
+	
+	protected SnmpValue getSnmpValue(final Optional<CollectionAttribute> target2) {
+		final SnmpValue value = SnmpUtils.getValueFactory().getOctetString(target2.get().getStringValue().getBytes());
+		return value;
+	}
+
+    protected boolean matchesValue(final String sourceType, final String sourceAlias, final String index, final CollectionAttribute a) {
```

Comment:
This method can be removed as it calls the `matches` method that already exist without altering signature.